### DISCUSSION TITLE
Accessing and changing a component in an already existing model of Ansys Twin Builder 2023 R2

### DISCUSSION THREAD

**Discussion by rubens1000** [ansys/pyaedt#4365] [Q&A]
Hello everyone,

I'm encountering an issue while working with Ansys TwinBuilder. I'm trying to access a specific component in my model and then modify one of its parameters or commenting out the component from the python script using pyaedt, but I seem to be having trouble finding the right approach.

Here is the code so far:

    from pyaedt import TwinBuilder
     from scipy.io import savemat
     # Initialize the Simplorer project
     # Make sure to adjust `specified_version` to match your ANSYS version
     simplorer= TwinBuilder(projectname=project_path, designname=design_name, specified_version="2023.2", non_graphical=True) 

    # Optionally, set up your simulation parameters here
    # Here I want to change one of the components in the model....

     # Run the simulation
      simplorer.analyze_nominal()
      report=simplorer.post.create_report(["Variable 1"],primary_sweep_variable="Time")
      solution = report.get_solution_data()
       plt = solution.plot(solution.expressions)  # Matplotlib axes object.   

       #Confirm that the simulation was completed and end script
       print("Simulation completed")

      simplorer.release_desktop(close_projects=True, close_desktop=True)

Thank you in advance for your help and guidance!

Best regards,
Ruben

**Comment by Samuelopez-ansys** [comment]
Hi @rubenidenABB ,

I have opened a random example from the Open Examples:
<img width="289" alt="image" src="https://github.com/ansys/pyaedt/assets/85613111/3fcded69-bf69-4864-a821-94939e6848ef">

You can access all components with app.modeler.schematic, for instance, you can modify the R value of the Resistor with ID 1 with the following script:

```
app = TwinBuilder(specified_version="2024.1")
r = app.modeler.schematic[1]

app.oeditor.ChangeProperty([
    "NAME:AllTabs",
    [
        "NAME:Quantities",
        [
            "NAME:PropServers",
            r.composed_name
        ],
        [
            "NAME:ChangedProps",
            [
                "NAME:R",
                "OverridingDef:=", True,
                "Value:=", "0.5ohm",
                "NetlistUnits:=", "ohm",
                "ShowPin:=", False,
                "Display:=", False,
                "Sweep:=", False,
                "SDB:=", False
            ]
        ]
    ]
])

```

As you can see, you can reuse the recorded script to do it.

**Reply by rubens1000** [reply]
Thanks for the reply! I am still encountering problems. I do not know how to find the ID of the component I want to access. Is there any way to find this based on the name of the components.

Here is the code I have tried with so far, but it does not seem to work:

c_name = "ES_CABLE10"
target_component = None
r = simplorer.modeler.schematic.components

for component in r:
    if component.name == c_name:
        target_component = component
        break

**Reply by rubens1000** [reply]
And as well, is there any place to see relevant examples like you did?

**Comment by Samuelopez-ansys** [comment]
@rubens1000  You can find examples here:
https://aedt.docs.pyansys.com/version/stable/examples/07-TwinBuilder/index.html

**Comment by Samuelopez-ansys** [comment]
@rubens1000 YOu just n eed:

```
c_name = "ES_CABLE10"
target_component = None
r = simplorer.modeler.schematic.components

for component in r:
    if r[component].name == c_name:
        target_component = component
        break

app.release_desktop()
```
